streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStandbyTasks.java

```diff
@@ -16,12 +16,42 @@
  */
 package org.apache.kafka.streams.processor.internals;
 
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.streams.processor.TaskId;
 
 class AssignedStandbyTasks extends AssignedTasks<StandbyTask> {
 
     AssignedStandbyTasks(final LogContext logContext) {
         super(logContext, "standby task");
     }
 
+    /**
+     * Closes tasks that have been reassigned at the end of a rebalance.
+     *
+     * @param revoked the tasks which are no longer owned
+     * @param revokedChangelogs the changelog topics of all tasks that were reassigned
+     */
+    void closeReassignedStandbyTasks(final Map<TaskId, Set<TopicPartition>> revoked, final List<TopicPartition> revokedChangelogs) {
+        for (final Map.Entry<TaskId, Set<TopicPartition>> entry : revoked.entrySet()) {
+            final TaskId taskId = entry.getKey();
+            if (!running.containsKey(taskId)) {
+                throw new IllegalStateException("Tried to close task but could not find it in assignment");
```

Comment:
nit: `Could not find the standby task {} while closing it` Read better as we immediately understand the issue is missing task